core/java/iesi-core/src/test/java/io/metadew/iesi/script/action/http/HttpExecuteRequestTest.java

```diff
@@ -0,0 +1,453 @@
+
+    }
+
+    @Test
+    void prepareOverrideHeaderAndNoQueries() throws HttpRequestBuilderException, URISyntaxException {
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        ActionParameter headersActionParameter = createActionParameter("headers", "Accept=application/xml");
+        Action action = createAction(requestActionParameter, headersActionParameter);
+        HttpHeader existingHeader = new HttpHeader("Accept", "application/json");
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter, headersActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(Collections.singletonList(existingHeader), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+        HttpRequestBase httpRequest = httpExecuteRequest.getHttpRequest().getHttpRequest();
+
+        assertThat(httpRequest.getAllHeaders()).hasSize(1);
+        assertThat(httpRequest.getFirstHeader("Accept").getValue()).isEqualTo("application/xml");
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"Accept=application/json", "Accept=application/json,X-API-KEY=12345"})
+    void prepareHeaderAndNoQueries(String headers) throws HttpRequestBuilderException, URISyntaxException {
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        ActionParameter headersActionParameter = createActionParameter("headers", headers);
+        Action action = createAction(requestActionParameter, headersActionParameter);
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter, headersActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+        HttpRequestBase httpRequest = httpExecuteRequest.getHttpRequest().getHttpRequest();
+
+        if (headers.equals("Accept=application/json")) {
+            assertThat(httpRequest.getAllHeaders()).hasSize(1);
+            assertThat(httpRequest.getFirstHeader("Accept").getValue()).isEqualTo("application/json");
+        } else if (headers.equals("Accept=application/json,X-API-KEY=12345")) {
+            assertThat(httpRequest.getAllHeaders()).hasSize(2);
+            assertThat(httpRequest.getFirstHeader("Accept").getValue()).isEqualTo("application/json");
+            assertThat(httpRequest.getFirstHeader("X-API-KEY").getValue()).isEqualTo("12345");
+        }
+
+
+    }
```

Comment:
why a parameterized unit test? it seems that for parameter 1 you already have the use case above, only parameter 2 (multiple headers) seems to be a applicable here

Reply:
SonarLint asks me to make a parameterized test because the tests were the same only the headers string changed.
The test above test the overriding of the current headers at component level, this test is when there are not headers at component level.

Reply:
Okay, but we have two use cases then?
1. 1 header pair provided
2. multiple (2) header pairs provided
These use cases both have a different outcome so they should be different unit tests.

having a if statement in your unit test marks that you're testing two different use cases in 1 unit test

Reply:
Okay I see, so I will create two different tests